src/audio/smart_amp/smart_amp.c

```diff
@@ -111,8 +111,11 @@ static ssize_t smart_amp_alloc_mod_memblk(struct smart_amp_data *sad,
 
 	/* query the required size from inner model. */
 	ret = mod->mod_ops->query_memblk_size(mod, blk);
-	if (ret <= 0)
+	if (ret <= 0) {
+		if (ret == 0)
+			ret = -EINVAL;
 		goto error;
+	}
 
 	/* allocate the memory block when returned size > 0. */
```

Comment:
I just revisited the design and the purpose should be that `ret = mod->mod_ops->query_memblk_size(mod, blk)` returns required memory size or negative errno. That is, `(ret == 0)` is allowed which states that no memory is required from inner model (of memblk `blk`). That makes sense because the current smart_amp design offers 3 memblk types to inner model for allocating buffers at different moments. Inner model doesn't have to request from all of them. (It depends on the design e.g. some can only know the required size of data buffer when init process is done, or when it's configured.)